python/sglang/srt/managers/schedule_batch.py

```diff
@@ -752,20 +752,24 @@ def prepare_for_extend(self):
         extend_lens = torch.tensor(self.extend_lens, dtype=torch.int32).to(
             self.device, non_blocking=True
         )
-        write_req_to_token_pool_triton[(bs,)](
-            self.req_to_token_pool.req_to_token,
-            self.req_pool_indices,
-            pre_lens,
-            self.seq_lens,
-            extend_lens,
-            self.out_cache_loc,
-            self.req_to_token_pool.req_to_token.shape[1],
-        )
-        # The triton kernel is equivalent to the following python code.
-        # self.req_to_token_pool.write(
-        #    (req.req_pool_idx, slice(pre_len, seq_len)),
-        #    out_cache_loc[pt : pt + req.extend_input_len],
-        # )
+        if global_server_args_dict["attention_backend"] != "torch_native":
```

Comment:
Why do you avoid the triton kernel here?

Reply:
I wanted to bypass triton when using torch_native attn backend, since some devices don't support triton currently (like HPU). But it also expands attention_backend flag's control scope , which maybe not a good practice. 
Any suggestions about this, do we want a dedicated flag to control the triton usage?

Reply:
probably okay for now.